### PR TITLE
Include read-only index versions in randomCompatibleVersion/randomCompatiblePreviousVersion

### PR DESCRIPTION
**Pull request by javanna**
Read-only versions are now included in all index versions, see #118793 . The next step is to broaden testing where possible to include index versions that cannot be written to. To do that, we change the behaviour of the existing randomCompatbileVersion and randomCompatiblePreviousVersion and methods to include read-only versions and we introduce corresponding variants of these methods that only return write compatible index versions.

As part of this change, we can also remove some of the `@UpdateForV9` annotations that relate to v7 index versions which randomCompatibleVersion no longer covered. That's now fixed and such tests can simply be restored.

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/bucket/GeoDistanceIT.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ protected boolean forbidPrivateIndexSettings() {
         return false;
     }
 
-    private final IndexVersion version = IndexVersionUtils.randomCompatibleVersion(random());
+    private final IndexVersion version = IndexVersionUtils.randomCompatibleWriteVersion(random());
```
Comment (javanna) [inline server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/bucket/GeoDistanceIT.java:59]: All of the IT tests create real indices, hence they can't use read-only versions.

File: modules/legacy-geo/src/test/java/org/elasticsearch/legacygeo/GeoJsonShapeParserTests.java [hunk 2]
```diff
@@ -344,8 +343,6 @@ public void testParsePolygon() throws IOException, ParseException {
         assertGeometryEquals(p, polygonGeoJson, false);
     }
 
-    @UpdateForV9(owner = UpdateForV9.Owner.SEARCH_ANALYTICS)
-    @AwaitsFix(bugUrl = "this test is using pre 8.0.0 index versions so needs to be removed or updated")
```
Comment (javanna) [inline modules/legacy-geo/src/test/java/org/elasticsearch/legacygeo/GeoJsonShapeParserTests.java:348]: This geo related tests and below are now functional again, and they should be kept in the codebase until v7 indices are supported.

**Review by drempapis — APPROVED** [review]
Thank you @javanna, LGTM

**Review by cbuescher — APPROVED** [review]
LGTM

**Review by iverase — APPROVED** [review]

**Pull request merged**